### ISSUE TITLE
Custom Edit Directions

### ISSUE THREAD

**Issue by prometx11** [pix2pixzero/pix2pix-zero#3]
In the description where you explain how to create new edit directions you mention to create 2 files with large number of sentences (~1000), but there is no explanation of what those sentences should be exactly. 

I tried to find out more details before opening this issue, but I could not figure out what the 2 files should contain exactly.

Thanks for the help and this amazing work!

**Comment by pix2pixzero** [comment]
Hi, 

The `src/make_edit_direction.py` script expects each sentence file to have one sentence per line. 
For instance, for the making a "banana" embedding, a "banana.txt" file should look something like this:

```
A ripe yellow banana sits on a surface, its curves contrasting with the straight lines around it.
The banana is the standout feature of the picture, its bright color drawing the eye.
The fruit's skin is slightly speckled, indicating its natural sweetness.
Shadows play across the surface of the banana, adding depth and dimension to the image.
The bright yellow color of the banana is a burst of happiness in an otherwise neutral background.
.
.
.

```

To generate these sentences, you could prompt your choice of language model (GPT3, ChatGPT, etc) with sentences like: 
"Generate many sentences that describe a picture containing a {word}."

Please let us know if you run into any issues with generating custom directions!


Regards,
Authors

**Comment by HelenMao** [comment]
@pix2pixzero Hi, thanks for your detailed instructions and releasing the codes of your inspiring work. I have tried "cat2yawning cat" by constructing text files following "Generate many sentences that describe a picture containing a yawning cat." using chatGPT. However, this editing direction nearly unalter the provided cat image and could not reproduce the results like the paper. Could you give more suggestions or release more editing direction files to verify the effectiveness of other editing directions?

**Comment by pix2pixzero** [comment]
Yes, we plan to release a huggingface demo and a large list of directions soon. 
Meanwhile, there are a couple of things you could try to improve the direction editing:

- scale the direction vector to make the direction more prominent. i.e. in [this file](https://github.com/pix2pixzero/pix2pix-zero/blob/main/src/edit_synthetic.py#L45) you can add a multiplier to the directions by changing `construct_direction(args.task_name)` to `construct_direction(args.task_name)*1.5`.

- You can decrease the amount of cross-attention guidance with the flag `--xa_guidance`. 

Please let us know if you still cannot get the results you want! 

Regards,
Authors

**Comment by rahulvigneswaran** [comment]
OpenAI's api keeps freezing and on top of that doing this for several directions seems expensive. Is there any other way to generate such sentences?

**Comment by rahulvigneswaran** [comment]
I have generated edit directions for 100 classes in cifar100 - https://github.com/rahulvigneswaran/pix2pix-zero-directions

**Comment by Larerr** [comment]
@rahulvigneswaran 404 Not Found. QAQ

**Comment by OrangeZz0331** [comment]
> Hi,
> 
> The `src/make_edit_direction.py` script expects each sentence file to have one sentence per line. For instance, for the making a "banana" embedding, a "banana.txt" file should look something like this:
> 
> ```
> A ripe yellow banana sits on a surface, its curves contrasting with the straight lines around it.
> The banana is the standout feature of the picture, its bright color drawing the eye.
> The fruit's skin is slightly speckled, indicating its natural sweetness.
> Shadows play across the surface of the banana, adding depth and dimension to the image.
> The bright yellow color of the banana is a burst of happiness in an otherwise neutral background.
> .
> .
> .
> ```
> 
> To generate these sentences, you could prompt your choice of language model (GPT3, ChatGPT, etc) with sentences like: "Generate many sentences that describe a picture containing a {word}."
> 
> Please let us know if you run into any issues with generating custom directions!
> 
> Regards, Authors

Thank you so much！
And I would like to konw if I want to try a style change, for example, if I need “sketch”style, how should I use chatgpt3.5 to generate sentences describing this style？

**Comment by israrbacha** [comment]
> I have generated edit directions for 100 classes in cifar100 - https://github.com/rahulvigneswaran/pix2pix-zero-directions

This link does not exist
